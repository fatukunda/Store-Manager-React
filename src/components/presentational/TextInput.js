import React from 'react';
import PropTypes from 'prop-types';

const TextInputField = ({ type, name, value, changed, required, className, id, placeholder }) => {
    return (
        <input type={type} name={name} id={id} value={value} onChange={changed} required={required} className={className} placeholder={placeholder} />
    );
};

TextInputField.propTypes = {
    type: PropTypes.string.isRequired,
    value: PropTypes.string,
    changed: PropTypes.func.isRequired,
    required: PropTypes.bool.isRequired
};
TextInputField.defaultProps = {
    value: ""
};

export default TextInputField;