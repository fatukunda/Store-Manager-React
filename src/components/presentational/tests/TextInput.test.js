import React from 'react';
import { shallow } from 'enzyme';
import TextInput from '../TextInput';
describe('TextInput', () => {
	it('should render correctly in "debug" mode', () => {
		const component = shallow(<TextInput debug />);

		expect(component).toMatchSnapshot();
	});
});
