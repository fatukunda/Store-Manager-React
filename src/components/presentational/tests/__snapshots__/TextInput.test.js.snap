// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
